src/chainstate/stacks/boot/cost-voting.clar

```diff
@@ -0,0 +1,106 @@
+;; The .cost-voting contract
+
+;; Error codes
+(define-constant ERR_PROPOSE 1)
+
+;; Cost vote token
+(define-fungible-token cost-vote-token)
+
+;; Proposal counter
+(define-data-var proposal-counter uint 0)
+
+;; Rollback proposal counter
+(define-data-var rollback-proposal-counter uint 0)
+
+;; cost-function proposals
+(define-map proposals
```

Comment:
Can you add another `define-map` for "confirmed" proposals and maps for tracking entries in it?

```scheme
(define-map confirmed-cost-functions
   ((cost-function-id uint))
   ((confirmed-function 
      (optional  
        {  function-contract: principal,
           function-name: (string-ascii 50),
           cost-function-contract: principal,
           cost-function-name: (string-ascii 50),
           confirmed-height: uint })))
(define-map functions-to-confirmed-ids
   ((function-contract principal) (function-name (string-ascii 50))
   ((cost-function-id uint)))
(define-data-var cost-function-max-id uint)
```

These data structures would allow us to iterate over the currently defined cost functions by reading from the map from `(0, cost-function-max-id)`. The reason the "optional" is required is to be able to delete a cost-function in O(1) by storing a `none`. The reason we need the `functions-to-confirmed-ids` map is that we'll want to check whether or not a definition already exists when confirming a proposal, and if so, overwrite rather than add a new entry.